api4/oauth_test.go

```diff
@@ -122,54 +108,22 @@ func TestUpdateOAuthApp(t *testing.T) {
 
 	updatedApp, resp := AdminClient.UpdateOAuthApp(oapp)
 	CheckNoError(t, resp)
-
-	if updatedApp.Id != oapp.Id {
-		t.Fatal("Id should have not updated")
-	}
-
-	if updatedApp.CreatorId != oapp.CreatorId {
-		t.Fatal("CreatorId should have not updated")
-	}
-
-	if updatedApp.CreateAt != oapp.CreateAt {
-		t.Fatal("CreateAt should have not updated")
-	}
-
-	if updatedApp.UpdateAt == oapp.UpdateAt {
-		t.Fatal("UpdateAt should have updated")
-	}
-
-	if updatedApp.ClientSecret != oapp.ClientSecret {
-		t.Fatal("ClientSecret should have not updated")
-	}
-
-	if updatedApp.Name != oapp.Name {
-		t.Fatal("Name should have updated")
-	}
-
-	if updatedApp.Description != oapp.Description {
-		t.Fatal("Description should have updated")
-	}
-
-	if updatedApp.IconURL != oapp.IconURL {
-		t.Fatal("IconURL should have updated")
-	}
+	require.Equal(t, updatedApp.Id, oapp.Id, "Id should have not updated")
+	require.Equal(t, updatedApp.CreatorId, oapp.CreatorId, "CreatorId should have not updated")
+	require.Equal(t, updatedApp.CreateAt, oapp.CreateAt, "CreateAt should have not updated")
+	require.NotEqual(t, updatedApp.UpdateAt, oapp.UpdateAt, "UpdateAt should have updated")
+	require.Equal(t, updatedApp.ClientSecret, oapp.ClientSecret, "ClientSecret should have not updated")
+	require.Equal(t, updatedApp.Name, oapp.Name, "Name should have updated")
+	require.Equal(t, updatedApp.Description, oapp.Description, "Description should have updated")
+	require.Equal(t, updatedApp.IconURL, oapp.IconURL, "IconURL should have updated")
```

Comment:
A lot of these would probably make more sense as `assert` rather than `require`. We generally only use `require` when we need to immediately fail a test (e.g. a required step failed or if something is nil it's going to cause a panic)

Reply:
+1 to what @jwilander suggests
also: expected before actual